x-pack/legacy/plugins/ml/public/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -907,7 +913,7 @@ export class TimeSeriesExplorer extends React.Component {
     timefilter.enableTimeRangeSelector();
     timefilter.enableAutoRefreshSelector();
 
-    this.subscriptions.add(timefilter.getTimeUpdate$().subscribe(this.refresh));
+    this.subscriptions.add(timefilter.getTimeUpdate$().subscribe(() => this.refresh(false)));
```

Comment:
Looks good overall to me. I'd like to give it an extra test locally, though. 
One question - would it be simpler to perhaps have a property on the class itself like `skipRefresh` and then in the callback sent to the subscription here set that property to `true` before starting the refresh?
```
this.subscriptions.add(timefilter.getTimeUpdate$().subscribe(() => {
  this.skipRefresh = true;
  this.refresh(false)
}));
```
Then you could pass that property value on as a prop to let the other components know a refresh is already happening and prevent a re-refresh.
I may be missing some context here.

Reply:
This would mean that we set `skipRefresh` to `true` for every `timefilter` update. But we need to set it programmatically and just want it to be enabled when a) the job selector flyout is opened and b) when a new job was selected and is in the process of updating single metric viewer. It must be set to `false` in other cases, otherwise updates triggered by the user when clicking the refresh button or changing the time range would'nt have any effect.